primap2/pm2io/_data_reading.py

```diff
@@ -41,6 +41,178 @@
 ]
 
 
+def convert_long_dataframe_if(
+    data_long: pd.DataFrame,
+    *,
+    coords_cols: Dict[str, str],
+    coords_defaults: Optional[Dict[str, Any]] = None,
+    coords_terminologies: Dict[str, str],
+    coords_value_mapping: Optional[Dict[str, Any]] = None,
+    coords_value_filling: Optional[Dict[str, Dict[str, Dict]]] = None,
+    filter_keep: Optional[Dict[str, Dict[str, Any]]] = None,
+    filter_remove: Optional[Dict[str, Dict[str, Any]]] = None,
```

Comment:
filter_keep and filter_remove don't make much sense here, I think. If I have a dataframe already, I would filter it separately before calling `convert_long_dataframe_if`, I think.

Reply:
My intention was to put all code that is the same for all long format dataframes into the function to keep custom scripts as short as possible and the functionality as close to the read_XXX functions as possbile

Reply:
Yeah, keeping the API as close as possible is a good reason, true. Individually, it might make a lot of sense for performance reasons to filter earlier, but users are free to do that and ignore the `filter_*` parameters, so it makes total sense to keep them.